library/time-library/pom.xml

```diff
@@ -51,4 +56,33 @@
         </dependency>
     </dependencies>
 
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>${shade.plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>deploy</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <shadedArtifactAttached>true
+                            </shadedArtifactAttached>
+                            <shadedClassifierName>deploy
+                            </shadedClassifierName>
+                            <artifactSet>
+                                <includes>
```

Comment:
I think there are a few more things you need to include here (as specified in the issue):

org.roaringbitmap:RoaringBitmap
uk.gov.gchq.gaffer:bitmap-library
uk.gov.gchq.gaffer:sketches-library

Once the shaded jar is compiled, please could you look at what is included and check it contains all the required dependencies for this time library.